### DISCUSSION TITLE
class doesn't work in Prod

### DISCUSSION THREAD

**Discussion by HeiShuXianSheng** [tailwindlabs/tailwindcss#6219] [Help] [answered]
**What version of Tailwind CSS are you using?**

"tailwindcss": "2.2.19"

**What build tool (or framework if it abstracts the build tool) are you using?**
"Next.js": "12.0.4",

**What version of Node.js are you using?**
v14.17.0

**What browser are you using?**
Chrome

**What operating system are you using?**
Windows

**Reproduction URL**
https://github.com/HeiShuXianSheng/next_tailwind

**Describe your issue**
everything is right when i use 'dev'.
but there is mistask when i use 'build',and then 'start'. In mode 'start', I visit localhost:3000, the Tailwind className not to work as expect, such as 'flex','h-full' doesn't work.

right:
![image](https://user-images.githubusercontent.com/57590348/143841940-398a367c-7be1-41cb-8dc8-5dbed8e3b209.png)

wrong:
![image](https://user-images.githubusercontent.com/57590348/143842044-a84b0f3f-a309-49c9-86c2-6c4abd8557f7.png)

**Comment by adamwathan** [comment, marked as answer]
Hey! Your `purge` configuration is incorrect so there are styles being removed in production that shouldn't be. It is only looking at `js` files but you are using `tsx` files for many of the pages/components in your project.

Update this line so that it actually matches the files you are using in your project:

https://github.com/HeiShuXianSheng/next_tailwind/blob/master/tailwind.config.js#L2

I'd also recommend moving to the [Just-in-Time engine](https://tailwindcss.com/docs/just-in-time-mode), it makes it easier to catch problems like this in development and is also faster with more features 👍🏻

**Reply by HeiShuXianSheng** [reply]
thanks for your reply，it‘s working fine now. I install tailwindcss follow the step of https://docs.tailwindchina.com/docs/guides/nextjs, So I didn’t notice the difference between the `js` & `tsx`.
